pkgs/development/yottadb/default.nix

```diff
@@ -0,0 +1,168 @@
+{
+  lib, gcc10Stdenv, fetchFromGitLab, pkgs, writeText
+}:
+
+gcc10Stdenv.mkDerivation rec {
+  pname = "yottadb";
+  version = "r1.34";
+
+  src = fetchFromGitLab {
+    owner = "YottaDB";
+    repo = "DB/YDB";
+    rev = "${version}";
+    leaveDotGit = true;
+    sha256 = "sha256-eNQz3xXG1xxaSfJWpkuLnxCKvCplaNbYlyawhNbMNd0=";
+  };
+
+  enableParallelBuilding = true;
+
+  nativeBuildInputs = with pkgs; [
+    file cmake gnumake curl tcsh
```

Comment:
OK, that is fine, but what if `stdenv.mkDerivation` was suddenly refactored and this automatic dependency was gone? Is it not better to specify all dependencies for each package explicitly, so it does not make any assumptions on anything else? If the dependency is already satisfied, there is no harm to it anyway, right?

Reply:
Maybe it would have been better to specify all dependencies for each package explicitly, but that's not how it's done right now and this won't change anytime soon, because it would basically mean having to change every single nix package.

The harm of adding random dependencies is that this might actually somehow break the build, e.g. when cross compiling plus it just adds code that, at best, does nothing